File: python/trigger_task.py
#!/usr/bin/env python3
"""Queue a Komos task run from Python.

Set KOMOS_API_KEY and KOMOS_TASK_ID before running:

    export KOMOS_API_KEY="sk_live_..."
    export KOMOS_TASK_ID="your-task-id"
    python python/trigger_task.py
"""

from __future__ import annotations

import json
import os
import sys
import urllib.error
import urllib.request


API_BASE_URL = 'https://api.komos.ai/public/v1'


def require_env(name: str) -> str:
    value = os.environ.get(name)
    if not value:
        print(f'Missing required environment variable: {name}', file=sys.stderr)
        sys.exit(1)
    return value


def main() -> None:
    api_key = require_env('KOMOS_API_KEY')
    task_id = require_env('KOMOS_TASK_ID')

    payload = {
        'inputs': {
            'example': 'hello from Python',
        },
        'clientRequestId': 'python-example-run',
    }

    request = urllib.request.Request(
        f'{API_BASE_URL}/tasks/{task_id}/runs',
        data=json.dumps(payload).encode('utf-8'),
        headers={
            'Authorization': f'Bearer {api_key}',
            'Content-Type': 'application/json',
        },
        method='POST',
    )

    try:
        with urllib.request.urlopen(request, timeout=30) as response:
            body = response.read().decode('utf-8')
            print(json.dumps(json.loads(body), indent=2))
    except urllib.error.HTTPError as exc:
        error_body = exc.read().decode('utf-8')
        print(f'Komos API returned HTTP {exc.code}: {error_body}', file=sys.stderr)
        sys.exit(1)


if __name__ == '__main__':
    main()

File: typescript/trigger-task.ts
const API_BASE_URL = "https://api.komos.ai/public/v1";

function requireEnv(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Missing required environment variable: ${name}`);
  }
  return value;
}

async function main() {
  const apiKey = requireEnv("KOMOS_API_KEY");
  const taskId = requireEnv("KOMOS_TASK_ID");

  const response = await fetch(`${API_BASE_URL}/tasks/${taskId}/runs`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${apiKey}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      inputs: {
        example: "hello from TypeScript",
      },
      clientRequestId: "typescript-example-run",
    }),
  });

  const body = await response.json();

  if (!response.ok) {
    throw new Error(`Komos API returned HTTP ${response.status}: ${JSON.stringify(body)}`);
  }

  console.log(JSON.stringify(body, null, 2));
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
